request_test.go

```diff
@@ -306,3 +310,78 @@ func TestDetectEncoding(t *testing.T) {
 		assert.Equal(t, tt.want, got, "detectEncoding(%+v)", tt.opts)
 	}
 }
+
+func TestNewRequestWithMetadata(t *testing.T) {
+	req := &transport.Request{Method: "foo"}
+	topts := TransportOptions{ServiceName: "bar"}
+	req, err := prepareRequest(req, nil, Options{TOpts: topts})
+	assert.NoError(t, err)
+	assert.Equal(t, "foo", req.Method)
+	assert.Equal(t, "bar", req.TargetService)
+}
+
+func TestNewRequestWithTransportMiddleware(t *testing.T) {
+	req := &transport.Request{Method: "foo"}
+	topts := TransportOptions{ServiceName: "bar"}
+	restore := transportmiddleware.Register(mockTransportMiddleware{})
+	defer restore()
+	req, err := prepareRequest(req, nil, Options{TOpts: topts})
```

Comment:
nit: For readability, when passing a naked nil as a position argument,
add a comment indicating the parameter name.

    prepareRequest(req, nil /* headers */, Options{...})